File: 10773.cpp
#include<bits/stdc++.h>
using namespace std;

int k;
stack<int> st;

int main() {
	cin >> k;
	
	while(k--) {
		int num;
		
		cin >> num;
		
		if(num != 0) {
			st.push(num);	
		}
		
		else {
			st.pop();
		}
	}
	
	int cnt = 0;
	
	while(!st.empty()) {
		cnt += st.top();
		st.pop();
	}
	
	cout << cnt << endl;
}

File: 10773.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Stack<Integer> s = new Stack<Integer>();
        int sum = 0;
        
        for(int i = 0; i < n; i++) {
            int num = sc.nextInt();
            
            if(num != 0) {
                s.push(num);
            }
            
            else {
                s.pop();
            }
        }
        
        while(!s.isEmpty()) {
            sum += s.peek();
            s.pop();
        }
        
        System.out.println(sum);
    }
}